qa/tasks/cephfs/test_misc.py

```diff
@@ -29,6 +29,19 @@ def test_statfs_on_deleted_fs(self):
         self.mount_a.umount_wait(force=True)
         p.wait()
 
+    def test_fuse_mount_on_already_mounted_path(self):
+        if not isinstance(self.mount_a, FuseMount):
+            self.skipTest("Require FUSE client")
+
+        # Try to mount already mounted path
+        # expecting EBUSY error
+        try:
+            self.mount_a.mount_wait()
```

Comment:
@nmshelke This calls https://github.com/ceph/ceph/blob/master/qa/tasks/cephfs/mount.py#L180 without sudo on a path that is already a ceph-fuse mount::

```
2022-02-05T20:52:02.502 DEBUG:teuthology.orchestra.run.smithi150:> mkdir -p -v /home/ubuntu/cephtest/mnt.0
2022-02-05T20:52:02.564 INFO:teuthology.orchestra.run:Running command with timeout 60
2022-02-05T20:52:02.565 DEBUG:teuthology.orchestra.run.smithi150:> chmod 0000 /home/ubuntu/cephtest/mnt.0
2022-02-05T20:52:02.585 DEBUG:teuthology.orchestra.run:got remote process result: 1
2022-02-05T20:52:02.586 INFO:teuthology.orchestra.run.smithi150.stderr:chmod: changing permissions of '/home/ubuntu/cephtest/mnt.0': Operation not permitted
```

Reply:
Added sudo for `chmod`

Reply:
Would skipping changing mode bits be a better choice?

Reply:
@vshankar Skipping `chmod` if path already mounted.

Reply:
I spent some time on this issue with Nikhil as well as alone last week and today. The issue is not in running chmod twice or multiple times.

```
mkdir somedir
chmod 0000 somedir
chmod 0000 somedir
```

But if we do change ownership chmod fails during second time -
```
mkdir somedir
chmod 0000 somedir
chown root:root somedir
chmod 0000 somedir # this will fail
```
So, IMO, the issue has to be with change of permissions.

This happens in case of teuthology tests too. chmod runs twice in teuthology, true, but between those two instances the ownership of the directory must have changed. I verified this by running a test last week. Following is a snippet  from the logs. This is where chmod is run prior to mounting -

```
2022-02-10T13:38:33.955 DEBUG:teuthology.orchestra.run.smithi198:> mkdir -p -v /home/ubuntu/cephtest/mnt.0
2022-02-10T13:38:33.960 INFO:teuthology.orchestra.run.smithi198.stdout:mkdir: created directory '/home/ubuntu/cephtest/mnt.0'
2022-02-10T13:38:33.961 DEBUG:teuthology.orchestra.run.smithi198:> ls -la /home/ubuntu/cephtest/mnt.0
2022-02-10T13:38:34.013 INFO:teuthology.orchestra.run.smithi198.stdout:total 8
2022-02-10T13:38:34.013 INFO:teuthology.orchestra.run.smithi198.stdout:drwxrwxr-x 2 ubuntu ubuntu 4096 Feb 10 13:38 .
2022-02-10T13:38:34.013 INFO:teuthology.orchestra.run.smithi198.stdout:drwxr-xr-x 5 ubuntu ubuntu 4096 Feb 10 13:38 ..
2022-02-10T13:38:34.013 INFO:tasks.cephfs.mount:total 8
drwxrwxr-x 2 ubuntu ubuntu 4096 Feb 10 13:38 .
drwxr-xr-x 5 ubuntu ubuntu 4096 Feb 10 13:38 ..

2022-02-10T13:38:34.014 INFO:teuthology.orchestra.run:Running command with timeout 60
2022-02-10T13:38:34.014 DEBUG:teuthology.orchestra.run.smithi198:> chmod 0000 /home/ubuntu/cephtest/mnt.0
2022-02-10T13:38:34.021 DEBUG:teuthology.orchestra.run.smithi198:> sudo modprobe fuse
2022-02-10T13:38:34.033 INFO:teuthology.orchestra.run:Running command with timeout 30
2022-02-10T13:38:34.034 DEBUG:teuthology.orchestra.run.smithi198:> sudo mount -t fusectl /sys/fs/fuse/connections /sys/fs/fuse/connections
2022-02-10T13:38:34.087 INFO:teuthology.orchestra.run.smithi198.stderr:mount: /sys/fs/fuse/connections: /sys/fs/fuse/connections already mounted or mount point busy.
2022-02-10T13:38:34.088 DEBUG:teuthology.orchestra.run:got remote process result: 32
2022-02-10T13:38:34.089 INFO:teuthology.orchestra.run:Running command with timeout 900
2022-02-10T13:38:34.089 DEBUG:teuthology.orchestra.run.smithi198:> ls /sys/fs/fuse/connections
2022-02-10T13:38:34.095 INFO:tasks.cephfs.fuse_mount:Pre-mount connections: []
2022-02-10T13:38:34.096 DEBUG:teuthology.orchestra.run.smithi198:> (cd /home/ubuntu/cephtest && exec sudo nsenter --net=/var/run/netns/ceph-ns--home-ubuntu-cephtest-mnt.0 sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage daemon-helper kill ceph-fuse -f --admin-socket '/var/run/ceph/$cluster-$name.$pid.asok' /home/ubuntu/cephtest/mnt.0 --id 0)
```

And following is where the chmod is executed after mounting Ceph FS -

```
2022-02-10T13:38:36.458 DEBUG:teuthology.orchestra.run.smithi198:> ls -la /home/ubuntu/cephtest/mnt.0
2022-02-10T13:38:36.513 INFO:teuthology.orchestra.run.smithi198.stdout:total 5
2022-02-10T13:38:36.513 INFO:teuthology.orchestra.run.smithi198.stdout:drwxrwxrwt 2 root   root      0 Feb 10 13:38 .
2022-02-10T13:38:36.514 INFO:teuthology.orchestra.run.smithi198.stdout:drwxr-xr-x 5 ubuntu ubuntu 4096 Feb 10 13:38 ..
2022-02-10T13:38:36.514 INFO:tasks.cephfs.mount:total 5
drwxrwxrwt 2 root   root      0 Feb 10 13:38 .
drwxr-xr-x 5 ubuntu ubuntu 4096 Feb 10 13:38 ..

2022-02-10T13:38:36.515 INFO:teuthology.orchestra.run:Running command with timeout 60
2022-02-10T13:38:36.515 DEBUG:teuthology.orchestra.run.smithi198:> chmod 0000 /home/ubuntu/cephtest/mnt.0
2022-02-10T13:38:36.521 DEBUG:teuthology.orchestra.run:got remote process result: 1
2022-02-10T13:38:36.522 INFO:teuthology.orchestra.run.smithi198.stderr:chmod: changing permissions of '/home/ubuntu/cephtest/mnt.0': Operation not permitted
```

The change of ownership must be due to the mounting command.

Reply:
Adding sudo fixes it, so I suppose this is the issue - http://pulpito.front.sepia.ceph.com/rishabh-2022-02-14_07:59:23-fs-wip-vshankar-testing-20220201-113815-distro-basic-smithi/

Reply:
> Adding sudo fixes it, so I suppose this is the issue - http://pulpito.front.sepia.ceph.com/rishabh-2022-02-14_07:59:23-fs-wip-vshankar-testing-20220201-113815-distro-basic-smithi/

`sudo ...` will surely fix it (anything for that matter :)

My comment about skipping doing mkdir/chmod for this test was more of a question. If adding `sudo` seems fine, I'm fine with that, but does it have any added advantage over skipping such checks?

Reply:
I missed out this comment. I don't think so. Both work fine, just that skipping would prevent running a redundant command.

Reply:
ACK.

Reply:
@nmshelke I'm ok with using sudo. Please switch to that. I'll pick this up in my testing branch.

Reply:
@vshankar Done